instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -354,9 +354,12 @@ def collect_request_attributes(
             result, path, path, query_string, sem_conv_opt_in_mode
         )
     if http_url:
-        _set_http_url(
-            result, remove_url_credentials(http_url), sem_conv_opt_in_mode
-        )
+        if _report_old(sem_conv_opt_in_mode) or (
+            sem_conv_opt_in_mode == _HTTPStabilityMode.HTTP_DUP
+        ):
+            _set_http_url(
```

Comment:
Did you check if the change will break other instrumentation that relies on asgi?

Reply:
I've tested the following cases and all passed without issue with the change
opentelemetry-instrumentation-asgi
opentelemetry-instrumentation-fastapi
opentelemetry-instrumentation-starlette
opentelemetry-instrumentation-django